earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
+
+    # Install extra debian packages we will need to support plugins.
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        graphviz
+
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    RUN poetry config installer.max-workers 10
```

Comment:
I'd like to know the rationale behind this one. The default is number of cores + 4. So:

- On a Github runner, it's 2 + 4 = 6. 
- On the remote satellite, it's 8 + 4 = 12.
- On the Hetzner machine, it's 32 + 4 = 36.

Notably, the default is _also_ the maximum value you can set. So for example, setting it to 10 on the Github runner is a no-op. For the current remote satellite, we're shaving off 2 cores. Why?

Reply:
If you don't, `poetry` throws warning about exceeding the number of workers, when workers exceed 10.  Specifically:

```
Connection pool is full, discarding connection: pypi.org. Connection pool size: 10
```

It throws that error multiple times, and in batches.  Which looks ugly and could make one think it's not working reliably.  I was getting that all the time running locally.

Reply:
Wow, that seems really broken 🤣 You would think they would configure the urllib3 pool to match the number of workers. Anyways, for these types of things, I find it helpful to include a comment on why it's being set, especially to help with blind copying/pasting.

Reply:
https://github.com/python-poetry/poetry/issues/8469